examples/example_edit_nwb.py

```diff
@@ -0,0 +1,32 @@
+import lindi
+import h5py
+import pynwb
+
+
+# Define the URL for a remote .zarr.json file
+url = 'https://kerchunk.neurosift.org/dandi/dandisets/000939/assets/11f512ba-5bcf-4230-a8cb-dc8d36db38cb/zarr.json'
+
+# Load the h5py-like client from the reference file system
+client = lindi.LindiH5pyFile.from_reference_file_system(url, mode='r+')
+
+# modify the age of the subject
+subject = client['general']['subject']  # type: ignore
+assert isinstance(subject, h5py.Group)
+del subject['age']  # type: ignore
+subject.create_dataset('age', data=b'3w')
+
+# Create a new reference file system
+rfs_new = client.to_reference_file_system()
+
+# Optionally write to a file
+# import json
+# with open('new.zarr.json', 'w') as f:
+#     json.dump(rfs_new, f)
+
+# Load a new h5py-like client from the new reference file system
+client_new = lindi.LindiH5pyFile.from_reference_file_system(rfs_new)
```

Comment:
Could the same result be achieved by editing the downloaded JSON directly, without going through h5py to delete the dataset and replace the dataset? I imagine this is easier for attributes, the values of which are stored in the JSON, but I imagine this can be done on datasets as well with careful handling. Going through h5py is convenient, but in a way, it adds unnecessary complexity because the data isn't an HDF5 file anymore; it's a JSON + references to chunked, compressed, and/or base64-encoded datasets.

Reply:
@rly I actually think going through h5py is the right way to do this because it uses zarr to make all the manipulations.

Reply:
I see. That makes sense.